integration/container/stop_test.go

```diff
@@ -110,3 +113,35 @@ func TestStopContainerWithTimeout(t *testing.T) {
 		})
 	}
 }
+
+// TestStopEventsOrder verifies that "stop" and "die" events are produced in the
+// correct order when stopping a container; see
+// https://github.com/moby/moby/pull/50136#discussion_r2152578561
+// https://github.com/docker/compose/pull/12950#issuecomment-2980759296
+func TestStopEventsOrder(t *testing.T) {
+	ctx := setupTest(t)
+	apiClient := testEnv.APIClient()
+
+	cID := container.Run(ctx, t, apiClient)
+
+	since := request.DaemonUnixTime(ctx, t, apiClient, testEnv)
+
+	err := apiClient.ContainerStop(ctx, cID, containertypes.StopOptions{})
+	assert.NilError(t, err)
+
+	var pollOpts []poll.SettingOp
+	if testEnv.DaemonInfo.OSType == "windows" {
+		pollOpts = append(pollOpts, poll.WithTimeout(StopContainerWindowsPollTimeout))
+	}
+
+	poll.WaitOn(t, container.IsStopped(ctx, apiClient, cID), pollOpts...)
```

Comment:
@vvoland added the poll; PTAL


We should probably look at that comment I left; really curious now!

Reply:
☝️ per discussion on Slack; @vvoland didn't necessarily mean an actual poll here, just making sure the timeout is long enough for windows; this can probably be an inspect again, if needed at all (will have a look)

Reply:
Yeah, that was my initial thinking behind setting it to 1 second; either it stops gracefully, or kill it after 1 second; both should produce `stop` -> `die` -> `exited` in that order.

Reply:
I guess, default timeout would be better (lesser code -> better code). Just need to check if respective image stops correctly (doesn't ignore signal used for stop of container created from image).